### PR TITLE
Wrote reference manual page on rules

### PR DESCRIPTION
**Pull request by mdgeorge4153**
Added a note to the documentation about parametric method invocations

### CODE REVIEW

File: docs/ref-manual/cvl/rules.md [hunk 1]
```diff
@@ -2,6 +2,15 @@ Rules
 =====
 
 Rules (along with {doc}`invariants`) are the main entry points for the Prover.
+A rule defines a sequence of [commands](statements) that should be executed
```
Comment (nd-certora) [inline docs/ref-manual/cvl/rules.md:5]: I'm not sure about using 'executed' here maybe 'simulated' or 'symbolically executed' or just 'checked'?

File: docs/ref-manual/cvl/rules.md [hunk 2]
```diff
@@ -25,23 +34,133 @@ params ::= cvl_type [ id ] { "," cvl_type [ id ] }
 Overview
```
Comment (nd-certora) [inline docs/ref-manual/cvl/rules.md:34]: do we have a placeholder for cvl_type? might be worth to add a link from the grammar

**Review by nd-certora — APPROVED** [review]
some comments that I'm not sure about - overall very good

File: docs/ref-manual/glossary.md [hunk 3]
```diff
@@ -29,6 +29,12 @@ quantified expression
   referred to as *quantified expressions*.  See {ref}`logic-exprs` for
   details about quantifiers in CVL.
 
+parametric rule
+  A parametric rule is a rule that calls an ambiguous method, either using a
+  method variable, or using an overloaded function name.  The prover will
+  generate a separate report for each possible instantiation of the method.
```
[the remark this answers is not in the document]
Comment (mdgeorge4153) [reply docs/ref-manual/glossary.md:35]: yes, I think so, although I can't remember where I learned that.

File: docs/ref-manual/cvl/rules.md (hunk 2 above)
Comment (mdgeorge4153) [reply docs/ref-manual/cvl/rules.md:34]: I have a separate PR where I add links for all the productions.  I think it's already merged into the master branch

Comment (mdgeorge4153) [reply docs/ref-manual/cvl/rules.md:34]: https://docs.certora.com/en/latest/docs/ref-manual/cvl/rules.html?highlight=cvl_type#syntax

File: docs/ref-manual/cvl/rules.md (hunk 1 above)
Comment (mdgeorge4153) [reply docs/ref-manual/cvl/rules.md]: changed to simulated

**Pull request merged**